src/NServiceBus.Core/Transports/Learning/LearningTransportInfrastructure.cs

```diff
@@ -43,23 +42,30 @@ public LearningTransportInfrastructure(SettingsHolder settings)
 
         public override OutboundRoutingPolicy OutboundRoutingPolicy { get; } = new OutboundRoutingPolicy(OutboundRoutingType.Unicast, OutboundRoutingType.Multicast, OutboundRoutingType.Unicast);
 
-        string FindSolutionRoot()
+        static string FindStoragePath()
         {
             var directory = AppDomain.CurrentDomain.BaseDirectory;
 
             while (true)
             {
+                // Finding a solution file takes precedence
                 if (Directory.EnumerateFiles(directory).Any(file => file.EndsWith(".sln")))
                 {
-                    return directory;
+                    return Path.Combine(directory, DefaultLearningTransportDirectory);
+                }
+
+                // When no solution file was found try to find a learning transport directory
+                if (Directory.EnumerateDirectories(directory).Any(dir => dir.Equals(DefaultLearningTransportDirectory, StringComparison.OrdinalIgnoreCase)))
```

Comment:
Seems like we shouldn't ignore case, because if you're on linux, paths are case-sensitive.